sdk/python/cmd/pulumi-language-python-exec

```diff
@@ -107,14 +108,38 @@ if __name__ == "__main__":
     logging.getLogger("asyncio").setLevel(logging.CRITICAL)
     exit_code = 1
     try:
+        # record the location of the user's program to return user tracebacks
+        user_program_abspath = os.path.abspath(os.path.join(args.PROGRAM, '__main__'))
+
         coro = pulumi.runtime.run_in_stack(lambda: runpy.run_path(args.PROGRAM, run_name='__main__'))
         loop.run_until_complete(coro)
         exit_code = 0
     except pulumi.RunError as e:
         pulumi.log.error(str(e))
     except Exception:
         # provide the user with traceback information for troubleshooting
-        error_msg = "Program failed with an unhandled exception:\n" + traceback.format_exc()
+        def get_user_stacktrace() -> str:
```

Comment:
Style nit, could this be please lifted into top-level `def` prefixed with `_` to indicate internal; Python supports infinite nesting but I'd love us to get away from that style, it's easy to abuse. The new definition should carry a doc comment explaining what it's trying to do as it's highly unusual.